examples/gagent-proxy-redundant-downstream-integration.py

```diff
@@ -0,0 +1,177 @@
+"""Juju-doctor probe for redundant downstream telemetry integrations.
+
+If cos-proxy is related to grafana-agent via the cos-agent endpoint, then cos-proxy and
+grafana-agent should not be related to the same "downstream-prometheus".
+
+This probe is a solution probe (not a charm probe) because it targets a cyclic relation between
+three charms.
+
+Context: As openstack incrementally transitioned from cos-proxy to grafana-agent, some deployments
+ended up with hybrid, invalid topologies.
+"""
+
+from typing import Dict, Optional
+
+import yaml
+
+
+def status(juju_statuses):
+    """Status assertion for a cyclic relation between cos-proxy, grafana-agent, and prometheus.
+
+    >>> status({"failing-status": test_invalid_status()})  # doctest: +ELLIPSIS
```

Comment:
- More explicit perhaps?
- The `test_` prefix is a convention for something slightly different.

```suggestion
    >>> status({"invalid-openstack-model": example_status_cyclic_gagent_cos_proxy()})  # doctest: +ELLIPSIS
```